ocaml-lsp-server/src/custom_requests/req_merlin_call_compatible.ml

```diff
@@ -0,0 +1,156 @@
+open Import
+
+let capability = ("handleMerlinCallCompatible", `Bool true)
+
+let meth = "ocamllsp/merlinCallCompatible"
+
+type params =
+  { uri : Uri.t
+  ; result_as_sexp : bool
+  ; command : string
+  ; args : string list
+  }
+
+let expected_params =
+  `Assoc
+    [ ("uri", `String "<DocumentUri>")
+    ; ("command", `String "<MerlinCommand>")
+    ; ("resultAsSexp?", `String "<true | false>")
+    ; ("args?", `String "<ListOfArguments>")
+    ]
+
+let as_sexp_from_structured_json params =
+  match List.assoc_opt "resultAsSexp" params with
+  | Some (`Bool value) -> value
+  | _ ->
+    (* If the parameter is incorrectly formatted or missing, it is assumed that
+       the result is not requested in the form of Sexp *)
+    false
+
+let command_from_structured_json params =
+  match List.assoc_opt "command" params with
+  | Some (`String command_name) -> Some command_name
+  | _ ->
+    (* If the parameter is incorrectly formatted or missing, we refuse to build
+       the parameter, [command] is mandatory. *)
+    None
+
+let uri_from_structured_json params =
+  match List.assoc_opt "uri" params with
+  | Some uri -> Some (Uri.t_of_yojson uri)
+  | _ ->
+    (* If the parameter is incorrectly formatted or missing, we refuse to build
+       the parameter, [command] is mandatory. *)
+    None
+
+let partition_args =
+  (* The function is relatively optimistic and attempts to treat literal data as
+     strings of characters. *)
+  function
+  | `String s -> Either.Left s
+  | `Bool b -> Either.Left (string_of_bool b)
+  | `Float f -> Either.Left (string_of_float f)
+  | `Int i -> Either.Left (string_of_int i)
+  | `Intlit i -> Either.Left i
+  | x -> Either.Right x
```

Comment:
The field `args` is already an array of strings: `"args": string[]`
Why do you need to parse it as if it was an array of heterogeneous values ?

Reply:
It is not, it is an array of `Yojson-ish` (ie: a list of `[> \`String of string]`).

Reply:
Right, but we could also consider invalid any other values than ``` `String ```.

I don't think it's wrong to accept heterogeneous values, but then it should be said in the spec:
````
```json
{
  "uri": DocumentUri,
  "command": string,
  "args": <string|bool|...>[],
  "resultAsSexp": boolean
}
```
````

I am just not sure that this is worth the added complexity...

Reply:
Ok, I'll probably change the specs

Reply:
(Specs changed)